### PR TITLE
Nevin/telemetry apt2

### PR DESCRIPTION
**Pull request by cnevinc**
To correctly list all events and map them to Amplitude, we need a programmatic/automatic way to list all the pings we have now ( instead of manually check the wiki)
This PR should fix the issue.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -230,7 +233,8 @@ dependencies {
     implementation "org.mozilla.components:ui-autocomplete:${Versions.android_components}"
 
     implementation "com.adjust.sdk:adjust-android:${Versions.adjust}"
-    implementation "com.google.android.gms:play-services-analytics:${Versions.firebase}" // Required by Adjust
+    implementation "com.google.android.gms:play-services-analytics:${Versions.firebase}"
```
Comment (mTwTm) [inline app/build.gradle]: ?

File: app/src/main/java/org/mozilla/focus/activity/MainActivity.java [hunk 2]
```diff
@@ -930,16 +930,6 @@ public HomeFragment createHomeScreen() {
         return fragment;
     }
 
-    @Override
```
Comment (mTwTm) [inline app/src/main/java/org/mozilla/focus/activity/MainActivity.java]: Why are we removing these two telemetry pings?

File: components/service/telemetry-compiler/build.gradle [hunk 3]
```diff
@@ -0,0 +1,14 @@
+apply plugin: 'java-library'
+
+
+sourceCompatibility = "7"
```
Comment (mTwTm) [inline components/service/telemetry-compiler/build.gradle]: Maybe 8?

File: components/service/telemetry-compiler/src/test/java/org/mozilla/telemetry/compiler/TelemetryAnnotationProcessorTest.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                                        "        extras = {@TelemetryExtra(name = \"a\", value = \"v\")})",
+                                "        public void send(){" +
+                                        "}",
+                                "}"
+                        )).generatedFiles();
+        assert (telemetryWrapper.size() > 0);
+    }
+
+    @org.junit.Test
+    public void getSupportedSourceVersion() {
```
Comment (mTwTm) [inline components/service/telemetry-compiler/src/test/java/org/mozilla/telemetry/compiler/TelemetryAnnotationProcessorTest.java]: Not sure if we need this.

File: components/service/telemetry-compiler/src/test/java/org/mozilla/telemetry/compiler/TelemetryAnnotationProcessorTest.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+                                        "        action = \"a\",\n" +
+                                        "        method = \"m\",\n" +
+//                                        "        object = \"o\",\n" +
+                                        "        value = \"v\",\n" +
+                                        "        extras = {@TelemetryExtra(name = \"a\", value = \"v\")})",
+                                "        public void send(){" +
+                                        "}",
+                                "}"
+                        )).generatedFiles();
+        assert (telemetryWrapper.size() > 0);
```
Comment (mTwTm) [inline components/service/telemetry-compiler/src/test/java/org/mozilla/telemetry/compiler/TelemetryAnnotationProcessorTest.java]: ==1?

File: components/service/telemetry-compiler/src/test/java/org/mozilla/telemetry/compiler/TelemetryAnnotationProcessorTest.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+                                        "        action = \"a\",\n" +
+                                        "        method = \"m\",\n" +
+                                        "        object = \"o\",\n" +
+                                        "        value = \"v\",\n" +
+                                        "        extras = {@TelemetryExtra(name = \"a\", value = \"v\")})",
+                                "        public void send(){" +
+                                        "}",
+                                "}"
+                        )).generatedFiles();
+        assert (telemetryWrapper.size() > 0);
```
Comment (mTwTm) [inline components/service/telemetry-compiler/src/test/java/org/mozilla/telemetry/compiler/TelemetryAnnotationProcessorTest.java]: ==1?

**Comment by mTwTm** [comment]
Please take a look at the Telemetry changing behavior comments:
1. We're removing two pings.
2. We're adding two ping entry points that might require version update.

File: app/src/main/java/org/mozilla/focus/activity/MainActivity.java (hunk 2 above)
Comment (cnevinc) [reply app/src/main/java/org/mozilla/focus/activity/MainActivity.java]: They are not used by the data analyst.

File: components/service/telemetry-compiler/src/test/java/org/mozilla/telemetry/compiler/TelemetryAnnotationProcessorTest.java (hunk 5 above)
Comment (cnevinc) [reply components/service/telemetry-compiler/src/test/java/org/mozilla/telemetry/compiler/TelemetryAnnotationProcessorTest.java]: fix

File: components/service/telemetry-compiler/src/test/java/org/mozilla/telemetry/compiler/TelemetryAnnotationProcessorTest.java (hunk 4 above)
Comment (cnevinc) [reply components/service/telemetry-compiler/src/test/java/org/mozilla/telemetry/compiler/TelemetryAnnotationProcessorTest.java]: fix

File: components/service/telemetry-compiler/src/test/java/org/mozilla/telemetry/compiler/TelemetryAnnotationProcessorTest.java (hunk 6 above)
Comment (cnevinc) [reply components/service/telemetry-compiler/src/test/java/org/mozilla/telemetry/compiler/TelemetryAnnotationProcessorTest.java]: fixed

File: components/service/telemetry-compiler/src/main/java/org/mozilla/telemetry/compiler/TelemetryAnnotationProcessor.java [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+        private static final int MAX_LENGTH_OBJECT = 20;
+        private static final int MAX_LENGTH_VALUE = 80;
+        private static final int MAX_EXTRA_KEYS = 200;
+        private static final int MAX_LENGTH_EXTRA_KEY = 15;
+        private static final int MAX_LENGTH_EXTRA_VALUE = 80;
+    }
+
+
+    @Override
+    public SourceVersion getSupportedSourceVersion() {
```
[the remark this answers is not in the document]
Comment (cnevinc) [reply components/service/telemetry-compiler/src/main/java/org/mozilla/telemetry/compiler/TelemetryAnnotationProcessor.java]: yes.  see: https://docs.oracle.com/javase/7/docs/api/javax/annotation/processing/AbstractProcessor.html#getSupportedSourceVersion()

File: components/service/TelemetryAnnotation/build.gradle [hunk 8]
```diff
@@ -0,0 +1,32 @@
+apply plugin: 'java-library'
+apply plugin: 'kotlin'
+
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    compile "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
+}
+
+sourceCompatibility = "7"
```
[the remark this answers is not in the document]
Comment (cnevinc) [reply components/service/TelemetryAnnotation/build.gradle]: fixed.

File: app/src/main/java/org/mozilla/focus/urlinput/UrlInputFragment.kt [hunk 9]
```diff
@@ -125,8 +117,12 @@ class UrlInputFragment : Fragment(), UrlInputContract.View, View.OnClickListener
             R.id.clear -> {
                 urlView.setText("")
                 urlView.requestFocus()
+                TelemetryWrapper.searchClear()
+            }
+            R.id.dismiss -> {
+                dismiss()
+                TelemetryWrapper.searchDismiss()
```
[the remark this answers is not in the document]
Comment (cnevinc) [reply app/src/main/java/org/mozilla/focus/urlinput/UrlInputFragment.kt]: fixed

File: app/src/main/java/org/mozilla/focus/urlinput/UrlInputFragment.kt [hunk 10]
```diff
@@ -125,8 +117,12 @@ class UrlInputFragment : Fragment(), UrlInputContract.View, View.OnClickListener
             R.id.clear -> {
                 urlView.setText("")
                 urlView.requestFocus()
+                TelemetryWrapper.searchClear()
```
[the remark this answers is not in the document]
Comment (cnevinc) [reply app/src/main/java/org/mozilla/focus/urlinput/UrlInputFragment.kt]: fixed

File: components/service/TelemetryAnnotation/build.gradle [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+targetCompatibility = "7"
+buildscript {
+    ext.kotlin_version = '1.2.71'
```
[the remark this answers is not in the document]
Comment (cnevinc) [reply components/service/TelemetryAnnotation/build.gradle]: > Please take special care with the Telemetry changing behavior comments:
> 
> 1. We're removing two pings.
I removed the two unused pings here cause I'm generating the first version of events.md. And I want it to be correct.
> 2. We're adding two ping entry points that might require version update.
done

File: app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt [hunk 12]
```diff
@@ -1090,10 +1092,12 @@ object TelemetryWrapper {
             method = Method.CANCEL,
             `object` = Object.SEARCH_BAR,
             value = "",
-            extras = [])
+            extras = [TelemetryExtra(name = Extra.VERSION, value = SEARCHCLEAR_TELEMETRY_VERSION)])
     @JvmStatic
     fun searchDismiss() {
-        EventBuilder(Category.ACTION, Method.CANCEL, Object.SEARCH_BAR).queue()
+        EventBuilder(Category.ACTION, Method.CANCEL, Object.SEARCH_BAR)
+                .extra(Extra.VERSION, SEARCHCLEAR_TELEMETRY_VERSION)
```
Comment (mTwTm) [inline app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt]: SearchDismiss using SEARCHCLEAR_TELEMETRY_VERSION?

Comment (cnevinc) [reply app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt]: fixed

**Pull request merged**